ratis-netty/src/main/java/org/apache/ratis/netty/client/DataStreamReplyDecoder.java

```diff
@@ -29,25 +29,22 @@
 
   @Override
   protected void decode(ChannelHandlerContext channelHandlerContext,
-                        ByteBuf byteBuf, List<Object> list) throws Exception {
+                        ByteBuf byteBuf, List<Object> list) {
 
     if(byteBuf.readableBytes() >= 24){
       long streamId = byteBuf.readLong();
       long dataOffset = byteBuf.readLong();
       long dataLength = byteBuf.readLong();
       if(byteBuf.readableBytes() >= dataLength){
-        DataStreamReplyImpl reply = new DataStreamReplyImpl(streamId,
+        DataStreamReplyByteBuffer reply = new DataStreamReplyByteBuffer(streamId,
             dataOffset,
             byteBuf.slice(byteBuf.readerIndex(), (int) dataLength).nioBuffer());
         byteBuf.readerIndex(byteBuf.readerIndex() + (int)dataLength);
         byteBuf.markReaderIndex();
         list.add(reply);
       } else {
         byteBuf.resetReaderIndex();
-        return;
```

Comment:
if byteBuf.readableBytes() >= dataLength is false, do we need to print err log ?

Reply:
This may be a good suggestion but is not related to this issue.  This is a refactoring but not try to change anything to the code.